File: core/src/main/java/io/snabble/sdk/ReceiptInfo.java
package io.snabble.sdk;

import java.util.Date;

public class ReceiptInfo {
    private String id;
    private String projectId;
    private long timestamp;
    private String pdfUrl;
    private String shopName;
    private String price;

    public ReceiptInfo(String id, String projectId, long timestamp, String pdfUrl, String shopName, String price) {
        this.id = id;
        this.projectId = projectId;
        this.timestamp = timestamp;
        this.pdfUrl = pdfUrl;
        this.shopName = shopName;
        this.price = price;
    }

    public String getId() {
        return id;
    }

    public String getProjectId() {
        return projectId;
    }

    @Deprecated
    public Date getDate() {
        return new Date(timestamp);
    }

    public long getTimestamp() {
        return timestamp;
    }

    public String getPdfUrl() {
        return pdfUrl;
    }

    public String getShopName() {
        return shopName;
    }

    public String getPrice() {
        return price;
    }
}

File: core/src/main/java/io/snabble/sdk/ReceiptInfo.kt
package io.snabble.sdk

/**
 * Class containing information about a order
 */
data class ReceiptInfo(
    /**
     * Unique id of the receipt / order
     */
    @JvmField
    val id: String,
    /**
     * Get the project id of the order
     */
    @JvmField
    val projectId: String,
    /**
     * Unix timestamp of the purchase
     */
    @JvmField
    val timestamp: Long,
    /**
     * Url to the pdf document, containing the receipt
     */
    @JvmField
    val pdfUrl: String?,
    /**
     * Name of the shop in which the order was fulfilled
     */
    @JvmField
    val shopName: String,
    /**
     * Final price of the order
     */
    @JvmField
    val price: String,

    /**
     * State of the order
     */
    @JvmField
    val isSuccessful: Boolean
)
